lsp-types/src/Language/LSP/Protocol/Types/Uri.hs

```diff
@@ -199,7 +199,13 @@ This is one of the most performance critical parts of HLS, do not
 modify it without profiling.
 -}
 data NormalizedFilePath = NormalizedFilePath !NormalizedUri {-# UNPACK #-} !Text
-  deriving stock (Generic, Eq, Ord)
+  deriving stock (Generic)
+
+instance Eq NormalizedFilePath where
+  NormalizedFilePath _uri1 fp1 == NormalizedFilePath _uri2 fp2 = fp1 == fp2
```

Comment:
Definitely needs a comment!
```suggestion
  -- We assume that filepath normalization does not change across the lifetime of HLS, so that
  -- if we have two `NormalizedFilePath`s with the same filepath, they also have the same 
  -- normalized URI.
  NormalizedFilePath _uri1 fp1 == NormalizedFilePath _uri2 fp2 = fp1 == fp2
```